easybuild/easyconfigs/s/SKA2/SKA2-0.3.7-GCCcore-12.2.0.eb

```diff
@@ -0,0 +1,32 @@
+easyblock = 'Binary'
```

Comment:
I'm not familiar with Rust/Cargo at all, so maybe this is a dumb question, but is it not possible to use the Cargo easyblock for this package?

Reply:
@bedroge it probably is. I'm in a similar position, and in fact have borrowed this from Simon's repo of as yet unsubmitted ECs anyway...

Reply:
@bedroge what do you think about proceeding as is and maybe moving to `Cargo` later?

Reply:
Would be okay with me, but just to familiarize myself a bit more with this Cargo easyblock a bit more, I just had a quick look at it. Running `python3 -m easybuild.easyblocks.generic.cargo .` should usually create the required list of crates, but it didn't work in the git checkout, because there was no `Cargo.lock`. However, the tarball from crates.io (https://crates.io/api/v1/crates/ska/0.3.7/download -> rename to .tar.gz) does have one, and that produced the list.

The following easyconfig worked for me, so I propose we use this?
```
easyblock = 'Cargo'

name = 'SKA2'
version = '0.3.7'

homepage = 'https://docs.rs/ska/latest/ska/'
description = """Split k-mer analysis (version 2) uses exact matching of split k-mer sequences to align closely
 related sequences, typically small haploid genomes such as bacteria and viruses."""

toolchain = {'name': 'GCCcore', 'version': '12.2.0'}

crates = [
    ('ska', version),
    ('adler', '1.0.2'),
    ('ahash', '0.8.11'),
    ('aho-corasick', '1.1.2'),
    ('allocator-api2', '0.2.16'),
    ('anstream', '0.6.13'),
    ('anstyle', '1.0.6'),
    ('anstyle-parse', '0.2.3'),
    ('anstyle-query', '1.0.2'),
    ('anstyle-wincon', '3.0.2'),
    ('anyhow', '1.0.81'),
    ('argmin', '0.9.0'),
    ('argmin-math', '0.3.0'),
    ('assert_fs', '1.1.1'),
    ('autocfg', '1.1.0'),
    ('bincode', '1.3.3'),
    ('bit-vec', '0.6.3'),
    ('bitflags', '1.3.2'),
    ('bitflags', '2.4.2'),
    ('bstr', '1.9.1'),
    ('buffer-redux', '1.0.1'),
    ('bytecount', '0.6.7'),
    ('byteorder', '1.5.0'),
    ('bytes', '1.5.0'),
    ('bzip2', '0.4.4'),
    ('bzip2-sys', '0.1.11+1.0.8'),
    ('cc', '1.0.90'),
    ('cfg-if', '1.0.0'),
    ('ciborium', '0.2.2'),
    ('ciborium-io', '0.2.2'),
    ('ciborium-ll', '0.2.2'),
    ('clap', '4.5.2'),
    ('clap_builder', '4.5.2'),
    ('clap_derive', '4.5.0'),
    ('clap_lex', '0.7.0'),
    ('colorchoice', '1.0.0'),
    ('colored', '2.1.0'),
    ('console', '0.15.8'),
    ('crc32fast', '1.4.0'),
    ('crossbeam-channel', '0.5.12'),
    ('crossbeam-deque', '0.8.5'),
    ('crossbeam-epoch', '0.9.18'),
    ('crossbeam-utils', '0.8.19'),
    ('crunchy', '0.2.2'),
    ('deranged', '0.3.11'),
    ('diff', '0.1.13'),
    ('difflib', '0.4.0'),
    ('dirs-next', '2.0.0'),
    ('dirs-sys-next', '0.1.2'),
    ('doc-comment', '0.3.3'),
    ('either', '1.10.0'),
    ('encode_unicode', '0.3.6'),
    ('equivalent', '1.0.1'),
    ('errno', '0.3.8'),
    ('fastrand', '2.0.1'),
    ('flate2', '1.0.28'),
    ('float-cmp', '0.9.0'),
    ('getrandom', '0.2.12'),
    ('globset', '0.4.14'),
    ('globwalk', '0.9.1'),
    ('half', '2.4.0'),
    ('hashbrown', '0.14.3'),
    ('heck', '0.4.1'),
    ('hermit-abi', '0.3.9'),
    ('ignore', '0.4.22'),
    ('indexmap', '2.2.5'),
    ('indicatif', '0.17.8'),
    ('instant', '0.1.12'),
    ('is-terminal', '0.4.12'),
    ('itertools', '0.10.5'),
    ('itoa', '1.0.10'),
    ('lazy_static', '1.4.0'),
    ('libc', '0.2.153'),
    ('libm', '0.2.8'),
    ('libredox', '0.0.1'),
    ('linux-raw-sys', '0.4.13'),
    ('log', '0.4.21'),
    ('lzma-sys', '0.1.20'),
    ('matrixmultiply', '0.3.8'),
    ('memchr', '2.7.1'),
    ('miniz_oxide', '0.7.2'),
    ('ndarray', '0.15.6'),
    ('needletail', '0.5.1'),
    ('noodles-bgzf', '0.26.0'),
    ('noodles-core', '0.14.0'),
    ('noodles-csi', '0.30.0'),
    ('noodles-tabix', '0.36.0'),
    ('noodles-vcf', '0.49.0'),
    ('normalize-line-endings', '0.3.0'),
    ('num-complex', '0.4.5'),
    ('num-conv', '0.1.0'),
    ('num-integer', '0.1.46'),
    ('num-traits', '0.2.18'),
    ('num_cpus', '1.16.0'),
    ('num_threads', '0.1.7'),
    ('number_prefix', '0.4.0'),
    ('once_cell', '1.19.0'),
    ('paste', '1.0.14'),
    ('percent-encoding', '2.3.1'),
    ('pkg-config', '0.3.30'),
    ('portable-atomic', '1.6.0'),
    ('powerfmt', '0.2.0'),
    ('ppv-lite86', '0.2.17'),
    ('predicates', '2.1.5'),
    ('predicates', '3.1.0'),
    ('predicates-core', '1.0.6'),
    ('predicates-tree', '1.0.9'),
    ('pretty_assertions', '1.4.0'),
    ('proc-macro2', '1.0.79'),
    ('quote', '1.0.35'),
    ('rand', '0.8.5'),
    ('rand_chacha', '0.3.1'),
    ('rand_core', '0.6.4'),
    ('rand_xoshiro', '0.6.0'),
    ('rawpointer', '0.2.1'),
    ('rayon', '1.9.0'),
    ('rayon-core', '1.12.1'),
    ('redox_syscall', '0.4.1'),
    ('redox_users', '0.4.4'),
    ('regex', '1.10.3'),
    ('regex-automata', '0.4.6'),
    ('regex-syntax', '0.8.2'),
    ('rustix', '0.38.31'),
    ('rustversion', '1.0.14'),
    ('ryu', '1.0.17'),
    ('same-file', '1.0.6'),
    ('serde', '1.0.197'),
    ('serde_derive', '1.0.197'),
    ('serde_json', '1.0.114'),
    ('similar', '2.4.0'),
    ('simple_logger', '4.3.3'),
    ('slog', '2.7.0'),
    ('slog-async', '2.8.0'),
    ('slog-json', '2.6.1'),
    ('slog-term', '2.9.1'),
    ('snap', '1.1.1'),
    ('snapbox', '0.4.17'),
    ('snapbox-macros', '0.3.8'),
    ('strsim', '0.11.0'),
    ('syn', '2.0.52'),
    ('take_mut', '0.2.2'),
    ('tempfile', '3.10.1'),
    ('term', '0.7.0'),
    ('termtree', '0.4.1'),
    ('thiserror', '1.0.58'),
    ('thiserror-impl', '1.0.58'),
    ('thread_local', '1.1.8'),
    ('time', '0.3.34'),
    ('time-core', '0.1.2'),
    ('time-macros', '0.2.17'),
    ('unicode-ident', '1.0.12'),
    ('unicode-width', '0.1.11'),
    ('utf8parse', '0.2.1'),
    ('version_check', '0.9.4'),
    ('walkdir', '2.5.0'),
    ('wasi', '0.11.0+wasi-snapshot-preview1'),
    ('winapi', '0.3.9'),
    ('winapi-i686-pc-windows-gnu', '0.4.0'),
    ('winapi-util', '0.1.6'),
    ('winapi-x86_64-pc-windows-gnu', '0.4.0'),
    ('windows-sys', '0.48.0'),
    ('windows-sys', '0.52.0'),
    ('windows-targets', '0.48.5'),
    ('windows-targets', '0.52.4'),
    ('windows_aarch64_gnullvm', '0.48.5'),
    ('windows_aarch64_gnullvm', '0.52.4'),
    ('windows_aarch64_msvc', '0.48.5'),
    ('windows_aarch64_msvc', '0.52.4'),
    ('windows_i686_gnu', '0.48.5'),
    ('windows_i686_gnu', '0.52.4'),
    ('windows_i686_msvc', '0.48.5'),
    ('windows_i686_msvc', '0.52.4'),
    ('windows_x86_64_gnu', '0.48.5'),
    ('windows_x86_64_gnu', '0.52.4'),
    ('windows_x86_64_gnullvm', '0.48.5'),
    ('windows_x86_64_gnullvm', '0.52.4'),
    ('windows_x86_64_msvc', '0.48.5'),
    ('windows_x86_64_msvc', '0.52.4'),
    ('xz2', '0.1.7'),
    ('yansi', '0.5.1'),
    ('zerocopy', '0.7.32'),
    ('zerocopy-derive', '0.7.32'),
]

builddependencies = [
    ('binutils', '2.39'),
    ('Rust', '1.75.0'),
]

sanity_check_paths = {
    'files': ['bin/ska'],
    'dirs': [],
}

sanity_check_commands = ["ska --help"]

moduleclass = 'bio'
```

Reply:
@verdurin ^

Reply:
I think it should be sufficient if we both think the current easyconfig is fine?

Reply:
I see the CI is complaining about missing checksums. I think I had the same issue as you mentioned on Slack (`--inject-checksums` not adding them), but it seems like Micket's trick works. Could you add them?